### PR TITLE
Initial code from oasis

### PR DESCRIPTION
**Pull request by cojmckee**

### CODE REVIEW

File: roles/content/tasks/1_manifest.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+- name: manifest | pull down manifest
```
Comment (evgeni) [inline roles/content/tasks/1_manifest.yml:1]: we have `theforeman.foreman.redhat_manifest` for that. would you want to switch to that now, or later?

File: roles/content/tasks/3_repo_sync.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  oasis_roles.satellite.repository_sync:
+    username: "{{ satellite_admin_username }}"
+    password: "{{ satellite_admin_password }}"
+    server_url: "{{ satellite_server_url }}"
+    product: "{{ item.product.name }}"
+    repository: "{{ item.name }}"
+    organization: "{{ satellite_organization }}"
+  loop: "{{ repositories.resources }}"
+  when: item.url  # Not all repositories have a URL
+  async: 999999
```
Comment (evgeni) [inline roles/content/tasks/3_repo_sync.yml]: fwiw, we recently removed that example from the docs, as it resulted in OOMed systems in a few cases (too many parallel syncs)

**Comment by evgeni** [comment]
please don't commit binaries (roles/install/files/python2-apypie-0.2.1-2.el7.noarch.rpm)

File: roles/rhsm/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# tasks file for rhsm
+- name: main | clean repos
+  import_tasks: 1_clean_repos.yml
```
Comment (evgeni) [inline roles/rhsm/tasks/main.yml]: the file is called 1_cleanrepos.yml

File: roles/rhsm/tasks/1_cleanrepos.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  find:
+    paths: /etc/yum.repos.d
+    patterns: "*.repo"
+  register: repo_files
+
+- name: Clean Repos | Remove files in /etc/yum.repos.d not redhat.repo
+  file:
+    path: "{{ item.path }}"
+    state: absent
+  when: item.path != "/etc/yum.repos.d/redhat.repo"
```
Comment (evgeni) [inline roles/rhsm/tasks/1_cleanrepos.yml]: this is not idempotent for disconnected setups

File: roles/rhsm/tasks/4_enable_repo.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#Enabling the repos
+- name: Enable Repos | Enabling the satellite repos
+  command: >-
```
Comment (evgeni) [inline roles/rhsm/tasks/4_enable_repo.yml]: why not `rhsm_repository` module?

**Comment by evgeni** [comment]
So a two thoughts:
* this contains a jenkins config -- can we split that out in a separate PR, merge that and let Jenkins at least run lint on the PR?
* I'm fine merging this, even if it does not comply with our self-imposed rules, but I think the binary file should go, and the `import_tasks` should use the right filename (accidental find, but that's why I'm asking about the linting in the above point)

File: ansible.cfg [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+ask_pass       = True
+#log_path       =/var/log/ansible.log
+inventory_ignore_extensions = ~, .swp, .md, .adoc, .csv, .retry, .pyc, .pyo, .example
+display_skipped_hosts = False
+deprecation_warnings = False
+command_warnings = False
+nocows         = 1
+retry_files_enabled = False
+host_key_checking = False
+collections_paths = /home/rsawyer/git/rh
```
Comment (kedark3) [inline ansible.cfg]: do we need this path ? it seems personal computer path.

File: ansible.cfg [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+[privilege_escalation]
+become         = False
+#become_method  = sudo
+#become_user    = root
+#become_ask_pass= True
+
+[ssh_connection]
+ssh_args = -o ControlMaster=auto -o ControlPersist=900s
+control_path = %(directory)s/%%h-%%r
+#pipelining = False
```
Comment (kedark3) [inline ansible.cfg]: is this left commented intentionally?

File: galaxy.yml [hunk 8]
```diff
@@ -0,0 +1,59 @@
+### REQUIRED
+
+# The namespace of the collection. This can be a company/brand/organization or product namespace under which all
+# content lives. May only contain alphanumeric characters and underscores. Additionally namespaces cannot start with
+# underscores or numbers and cannot contain consecutive underscores
+namespace: oasis_roles
```
Comment (kedark3) [inline galaxy.yml:6]: will this remain OASIS or RedHatSatellite?

File: roles/config/meta/main.yml [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  # versions.
+  # If you don't wish to enumerate all versions for a particular platform, use
+  # 'all'.
+  # To view available platforms and versions (or releases), visit:
+  # https://galaxy.ansible.com/api/v1/platforms/
+  #
+  platforms:
+    - name: Fedora
+      versions:
+        - 7.7
```
Comment (kedark3) [inline roles/config/meta/main.yml:37]: I think this number should be different. Fedora 7.7 wouldn't be right. Do you mind updating it?

File: roles/config/tasks/2_lifecycle_env.yml [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+  oasis_roles.satellite.lifecycle_environment:
+    username: "{{ satellite_admin_username }}"
+    password: "{{ satellite_admin_password }}"
+    server_url: "{{ satellite_server_url }}"
+    name: "{{ satellite_prod_lifecycle_env }}"
+    label: "{{ satellite_prod_lifecycle_env }}"
+    description: "{{ satellite_prod_lifecycle_env }}"
+    prior: "Library"
+    organization: "{{ satellite_organization }}"
+    state: "present"
```
Comment (kedark3) [inline roles/config/tasks/2_lifecycle_env.yml]: should we make `state` a variable?

File: roles/config/tasks/1_host_collections.yml [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+### host collections
+- name: "config host collections | create PKI host collections"
+  oasis_roles.satellite.host_collection:
+    username: "{{ satellite_admin_username }}"
+    password: "{{ satellite_admin_password }}"
+    server_url: "{{ satellite_server_url }}"
+    name: "{{ item.name }}"
+    description: "{{ item.description }}"
+    organization: "{{ satellite_organization }}"
+    state: present
```
Comment (kedark3) [inline roles/config/tasks/1_host_collections.yml]: should we make `state` a variable with default value?

File: ansible.cfg (hunk 6 above)
Comment (bherrin3) [inline ansible.cfg]: Would expect project specific relative path:

```suggestion
collections_paths = ./collections
```

File: roles/config/tasks/3_content_views.yml [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+- name: "config content views | Create a composite content view"
+  oasis_roles.satellite.content_view:
+    username: "{{ satellite_admin_username }}"
+    password: "{{ satellite_admin_password }}"
+    server_url: "{{ satellite_server_url }}"
+    name: "{{ item.name }}"
+    organization: "{{ satellite_organization }}"
+    composite: true
+    auto_publish: true
```
Comment (kedark3) [inline roles/config/tasks/3_content_views.yml]: `composite` and `auto_publish` can be made vars with default of `true`?

File: roles/config/tasks/4_activation_keys.yml [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+    password: "{{ satellite_admin_password }}"
+    server_url: "{{ satellite_server_url }}"
+    name: "{{ item.name }}"
+    organization: "{{ satellite_organization }}"
+    lifecycle_environment: "{{ item.lifecycle_environment }}"
+    content_view: "{{ item.content_view }}"
+    host_collections: "{{ item.host_collections }}"
+    subscriptions: "{{ item.subscriptions }}"
+    content_overrides: "{{ item.content_overrides }}"
+    #auto_attach: "{{ item.auto_attach }}"
```
Comment (kedark3) [inline roles/config/tasks/4_activation_keys.yml]: Reason to leave these commented out?

File: roles/config/tasks/5_subnets.yml [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+    template_proxy: "{{ item.template_proxy | default(omit) }}"
+    vlanid: "{{ item.vlanid | default(omit) }}"
+    mtu: "{{ item.mtu | default(omit) }}"
+    domains: "{{ item.domains }}"
+    organizations: "{{ item.organizations }}"
+    locations: "{{ item.locations }}"
+    server_url: "{{ satellite_server_url }}"
+    username: "{{ satellite_admin_username }}"
+    password: "{{ satellite_admin_password }}"
+    state: present
```
Comment (kedark3) [inline roles/config/tasks/5_subnets.yml]: same about state here .

File: roles/config/tasks/6_domains.yml [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+  oasis_roles.satellite.domain:
+    username: "{{ satellite_admin_username }}"
+    password: "{{ satellite_admin_password }}"
+    server_url: "{{ satellite_server_url }}"
+    name: "{{ item.name }}"
+    description: "{{ item.description | default(omit) }}"
+    locations: "{{ item.locations }}"
+    organizations: "{{ item.organizations }}"
+    dns_proxy: "{{ satellite_fqdn | default(omit) }}"
+    state: present
```
Comment (kedark3) [inline roles/config/tasks/6_domains.yml]: same q about `state` being a variable here as well.

File: roles/config/tasks/7_hostgroups.yml [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+    operatingsystem: "{{ item.operatingsystem | default(omit) }}"
+    medium: "{{ item.medium | default(omit) }}"
+    ptable: "{{ item.ptable | default(omit) }}"
+    root_pass: "{{ item.root_pass | default(omit) }}"
+    compute_resource: "{{ item.compute_resource | default(omit) }}"
+    compute_profile: "{{ item.compute_profile | default(omit) }}"
+    content_view: "{{ item.content_view | default(omit) }}"
+    puppetclasses: "{{ item.puppetclasses | default(omit) }}"
+    parameters: "{{ item.parameters | default(omit) }}"
+    state: present
```
Comment (kedark3) [inline roles/config/tasks/7_hostgroups.yml]: same q about `state` being a variable here as well.

File: roles/install/handlers/main.yml [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+  command: satellite-installer
+
+- name: firewalld_reload
+  systemd:
+    name: firewalld
+    state: reloaded
+
+- name: run_installer
+  command: >-
+    satellite-installer --scenario satellite
```
Comment (bherrin3) [inline roles/install/handlers/main.yml:11]: 1. Is there any reason why the satellite installer below is hardcoded to `satellite` (what about capsule or other scenarios)?
2. Why are we using this and not deferring to a source `foreman_installer` task or dev source installer?
* https://github.com/theforeman/forklift/blob/master/roles/foreman_installer/tasks/install.yml

```suggestion
    satellite-installer --scenario {{ satellite_scenario }}
```

File: roles/content/tasks/2_enablerepos.yml [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+
+- name: enable repos | RHEL repositories with label that need releasever
+  oasis_roles.satellite.repository_set:
+    username: "{{ satellite_admin_username }}"
+    password: "{{ satellite_admin_password }}"
+    server_url: "{{ satellite_server_url }}"
+    organization: "{{ satellite_organization }}"
+    label: "{{ item.label }}"
+    repositories: "{{ item.repos }}"
+    state: enabled
```
Comment (kedark3) [inline roles/content/tasks/2_enablerepos.yml:16]: same q about `state` being a variable here as well.

File: roles/rhsm/tasks/4_enable_repo.yml (hunk 5 above)
Comment (bherrin3) [reply roles/rhsm/tasks/4_enable_repo.yml]: Could implement the module in a role with something like:

```
  rhsm_repository:
    name:  "{{ item.name }}"
    state: "{{ item.state | default('present') }}"
    purge: "{{ item.purge | default('no') }}"
  loop: "{{ repos }}"
```

And then make the direct call to that role with something like:

```
         - role: rhsm-repository
           repos:
            - rhel-{{ ansible_distribution_major_version }}-server-satellite-{{ satellite_version }}-rpms
            - rhel-{{ ansible_distribution_major_version }}-server-rpms
```

**Comment by ehelms** [comment]
As people comment and review, I want to remind that the general idea we had was to bring in these large contributions and then make adjustments. There is balance to strike with reviewing and @cojmckee making changes with filing Github issues to address later. I'll leave a large part of that decision up to @cojmckee on how much he wants to handle reviews here vs follow up.

**Review by kedark3 — COMMENTED** [review]
Thanks Cory for our 1st PR with code in this repo. I have few comments, mostly nit pics maybe and easy to address. Let me know what you think.

**Comment by kedark3** [comment]
I agree with @ehelms on his assertion between review and follow up.

File: roles/content/tasks/1_manifest.yml (hunk 1 above)
Comment (cojmckee) [reply roles/content/tasks/1_manifest.yml:1]: I will upload the new code later

File: roles/install/handlers/main.yml (hunk 17 above)
Comment (cojmckee) [reply roles/install/handlers/main.yml:11]: Have a capsule role that will be uploaded

File: ansible.cfg (hunk 6 above)
Comment (cojmckee) [reply ansible.cfg]: This file has been deleted

Comment (cojmckee) [reply ansible.cfg]: This file has been deleted

File: galaxy.yml (hunk 8 above)
Comment (cojmckee) [reply galaxy.yml:6]: in RedHatSatellite

File: roles/content/tasks/3_repo_sync.yml (hunk 2 above)
Comment (cojmckee) [reply roles/content/tasks/3_repo_sync.yml]: removed from code @evgeni

File: roles/rhsm/tasks/main.yml (hunk 3 above)
Comment (cojmckee) [reply roles/rhsm/tasks/main.yml]: Thanks and fixed @evgeni

File: roles/rhsm/tasks/1_cleanrepos.yml (hunk 4 above)
Comment (cojmckee) [reply roles/rhsm/tasks/1_cleanrepos.yml]: added a check for that file @evgeni

File: ansible.cfg (hunk 7 above)
Comment (cojmckee) [reply ansible.cfg]: This file has been removed @kedark3

File: roles/config/tasks/2_lifecycle_env.yml (hunk 10 above)
Comment (cojmckee) [reply roles/config/tasks/2_lifecycle_env.yml]: Change this to state into a variable

File: roles/config/tasks/4_activation_keys.yml (hunk 13 above)
Comment (cojmckee) [reply roles/config/tasks/4_activation_keys.yml]: removed the comment out code

File: roles/config/tasks/5_subnets.yml (hunk 14 above)
Comment (cojmckee) [reply roles/config/tasks/5_subnets.yml]: change this

File: roles/config/tasks/6_domains.yml (hunk 15 above)
Comment (cojmckee) [reply roles/config/tasks/6_domains.yml]: same here has been changed

File: roles/content/tasks/2_enablerepos.yml (hunk 18 above)
Comment (cojmckee) [reply roles/content/tasks/2_enablerepos.yml:16]: changed here to a var

File: roles/config/tasks/3_content_views.yml (hunk 12 above)
Comment (cojmckee) [reply roles/config/tasks/3_content_views.yml]: created satellite_auto_publish var

File: roles/config/tasks/1_host_collections.yml (hunk 11 above)
Comment (cojmckee) [reply roles/config/tasks/1_host_collections.yml]: changed this to a var

File: roles/config/tasks/7_hostgroups.yml (hunk 16 above)
Comment (cojmckee) [reply roles/config/tasks/7_hostgroups.yml]: changed this to a var

File: roles/config/meta/main.yml (hunk 9 above)
Comment (cojmckee) [reply roles/config/meta/main.yml:37]: changed

File: roles/install/handlers/main.yml (hunk 17 above)
Comment (cojmckee) [reply roles/install/handlers/main.yml:11]: changed the var to a variable

File: roles/rhsm/tasks/4_enable_repo.yml (hunk 5 above)
Comment (cojmckee) [reply roles/rhsm/tasks/4_enable_repo.yml]: added the rhsm_repository module to the play

**Pull request merged**